src/pages/home/report/ReportActionsView.js

```diff
@@ -218,6 +219,15 @@ class ReportActionsView extends React.Component {
             });
         }
 
+        // If the last unread message was deleted, remove the *New* green marker and the *New Messages* notification at scroll just as the deletion starts.
+        if (this.props.reportActions.length > 0 && this.props.reportActions[0].pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE && !this.props.network.isOffline) {
```

Comment:
I've added `!this.props.network.isOffline` in order to have a consistent behavior while offline. In *current staging*, while offline, the **New** markers do not disappear. In this PR, without this check, while offline, the **New** markers would disappear only for the last message. See this video:  

https://github.com/Expensify/App/assets/56603839/6a827893-b7d2-4ac3-a07e-792e51167761

Reply:
https://github.com/Expensify/App/assets/25520267/7863306c-8412-4ff7-88cc-5fa5bfbbd0b7

I think the condition can be improved. Right now, it's called twice.

Reply:
I. I've added a new condition in order to prevent *unnecessary re-renders*.  

II. I'm thinking which is the impact of the if executing twice.
1. Could you add `const now = Date.now()` first line after the `if condition` and `console.log('Elapsed: ', Date.now() - now)` before the new `if (newMarkerReportActionID !== this.state.newMarkerReportActionID)` introduced and test against the chat which has the most messages for you?
2. I've sent some large PDFs, images and messages in a chat. I get `1, 2 or 3ms` for the processing time, which should not be an issue.
3. If we still want to prevent the `condition` I may do something like creating a new class variable `this.isActionPendingDelete = false` and update the original condition as follows:
```js
if (this.props.reportActions.length > 0 && !this.props.network.isOffline) {
  if (this.props.reportActions[0].pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE && this.isActionPendingDelete !== true) {
      this.isActionPendingDelete = true;
      // Same logic here
  } else if (this.props.reportActions[0].pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE && this.isActionPendingDelete === true) {
      this.isActionPendingDelete = false;
  }
}
```

I would like to know your opinion about `II`.

Reply:
I've also added now `ReportUtils.isUnread(this.props.report)` inside the condition, such that is does not return `true` anymore when deleting the last message if it(or any other one) was NOT marked as unread.

Reply:
@mollfpr What's your opinion on this?